docs/specifications/tests/DNSSEC-TP/dnssec16.md

```diff
@@ -106,40 +115,52 @@ DS16_MIXED_DELETE_CDS                | ERROR   | "Delete" CDS record is mixed wi
 
 7.  For each name server IP in the *CDS RRsets* set do:
 
-    1. If the CDS RRset is non-empty, get the RRset and its RRSIG
-       records, else go to next name server IP address. The set of the RRSIG
-       records may be empty.
-    2. If any CDS record is a "delete" CDS, then do:
+    1. If the CDS RRset is empty go to next name server IP address.
+    2. Get the CDS RRset and its RRSIG records from the *CDS RRsets* set
+       for the name server IP. The RRSIG records may be absent.
```

Comment:
Yes, it means that the RRSIG records may be absent, there might be no RRSIG records for the CDS RRset.

No, at 2 the CDS RRset is non-empty.

RRSIG records are special. They do not constitute an RRset. They are a kind of attachments to the RRset that they sign. Here the RRset is the CDS RRset, and above in 5.ii.d (as md is rendered) the RRset and the associated RRSIG records are collected. It is possible that we have an CDS RRset but no RRSIG records for the RRset. If there is no CDS RRset we do not fetch any RRSIG records for the non-existent RRset.

I do not want to call it an RRSIG set, because that would sound like a RRSIG RRset, which it is not.

If you do see it as a set, then I do not see the difference between an empty set and a absent set.

Reply:
> RRSIG records are special. They do not constitute an RRset. They are a kind of attachments to the RRset that they sign.

Thanks for explaining! I wasn't aware of that.

> If you do see it as a set, then I do not see the difference between an empty set and a absent set.

If the set is absent that means there is no set. The same way having the number zero is different from having no number and having the empty string is different not having a string.

The new wording is good from a programmer's perspective.